In [1]:
import nltk
from pyspark import SparkConf, SparkContext
from pyspark.sql import SQLContext, SparkSession
from nltk.corpus import stopwords
import string

In [2]:
def word_tokenize(line):
	if isinstance(line, str):
		return nltk.word_tokenize(line)
	else :
		return nltk.word_tokenize(str(line))

In [3]:
def pos_tag(word):
#     nltk.download('averaged_perceptron_tagger')
    return nltk.pos_tag([word])

In [4]:
def handle_csv(path):
#     'hdfs://cluster001:9000/user/input/test.csv'
	data_frame_text = spark_session.read.csv(path)

	# keep the useful columns
	data_frame_text = data_frame_text.select('_c2','_c3','_c4')
	# convert to rdd
	rdd_text = data_frame_text.rdd.map(lambda x: [str(x[0]), x[1], x[2]])

	# remove the first line
	header_text = rdd_text.first()
	rdd_text_noheader = rdd_text.filter(lambda row: row != header_text)
	return rdd_text_noheader

In [13]:
def containsAny(seq,aset):
    for c in seq:
         if c in aset:
                return True
    return False

In [5]:
conf = SparkConf().setMaster("yarn").setAppName("BDA_TP_1_b")
sc = SparkContext(conf = conf)
spark_session = SparkSession.builder.getOrCreate()

In [57]:
!wget https://query.data.world/s/pqtxip4c4etmpusphzqfh33dnjgs73 -O abc_news_86680728811
!wget https://query.data.world/s/rdaaxhfhqbga3pmxkibvvmae2osrri -O bbc_228735667216
!wget https://query.data.world/s/vduiyet3zw3k5ugdalpm6pqa7s5aur -O cbs_news_131459315949
!wget https://query.data.world/s/majoz3zwxo7bopmycyqyuitjidqiym -O cnn_5550296508
!wget https://query.data.world/s/x5oxdbb7pvvp6gvaam5qiovceplxfd -O fox_and_friends_111938618893743
!wget https://query.data.world/s/k3gh6ldxoxo67gqyn6cwruju32q4cs -O fox_news_15704546335
!wget https://query.data.world/s/g6v33jwtht2awapnqmwsmryzdfexfp -O nbc_news_155869377766434
!wget https://query.data.world/s/aqu22wxid2txc3ug4bgkramejeb44h -O npr_10643211755
!wget https://query.data.world/s/27m3ayqwtjyeyucq3vvcz3cohb7c5s -O the_huffington_post_18468761129
!wget https://query.data.world/s/rq2njove26gq54jexuowouwx5kimbh -O the_los_angeles_times_5863113009

!wget https://query.data.world/s/cmpzvuydrxmztm7ah7ofbbm23ndo2s -O the_new_york_times_5281959998
!wget https://query.data.world/s/nuryoed6pjv35eikrfv5e7yf47twfg -O the_wall_street_journal_8304333127
!wget https://query.data.world/s/fnxrwmfgnojbdepznraakl64uunpz5 -O the_washington_post_6250307292
!wget https://query.data.world/s/jy6m2wmciyjf62pl4x2qaf6mph52yd -O time_10606591490
!wget https://query.data.world/s/iknrnrrza7rmjwmx7epwheh4uen4fk -O usa_today_13652355666

--2020-03-24 15:24:19--  https://query.data.world/s/pqtxip4c4etmpusphzqfh33dnjgs73
Resolving query.data.world (query.data.world)... 18.210.128.19, 35.169.251.217, 107.23.216.235
Connecting to query.data.world (query.data.world)|18.210.128.19|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://download.data.world/file_download/martinchek/2012-2016-facebook-posts/abc-news-86680728811.csv%2Fabc-news-86680728811.csv?auth=eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJwcm9kLXVzZXItY2xpZW50OnFpYW5oZW5nIiwiaXNzIjoiYWdlbnQ6cWlhbmhlbmc6OmUwN2U0NmE4LWRlMjQtNGZlZi1hMDQ3LTBlYjUzZGE3MTNjZSIsImlhdCI6MTU4NTAzMDAxOCwicm9sZSI6WyJ1c2VyIiwidXNlcl9hcGlfYWRtaW4iLCJ1c2VyX2FwaV9yZWFkIiwidXNlcl9hcGlfd3JpdGUiXSwiZ2VuZXJhbC1wdXJwb3NlIjpmYWxzZSwidXJsIjoiN2Y4ZWIxM2VjNmYwYjE1ZDMzNWQxZjU4ZWFiY2YwNWQ3YWQwZWM1MSJ9.iD-uIJpICUySQ0KJtglYtGC0sX11YaHmJywtFD1CvRpLpbFso0TIV-hfEqf7lbSvj5KkshJFLu-RkF8LPi2lTA [following]
--2020-03-24 15:24:21--  https://download.data.world/file_download/martinch

Resolving query.data.world (query.data.world)... 107.23.216.235, 35.169.251.217, 18.210.128.19
Connecting to query.data.world (query.data.world)|107.23.216.235|:443... failed: Connection refused.
Connecting to query.data.world (query.data.world)|35.169.251.217|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://download.data.world/file_download/martinchek/2012-2016-facebook-posts/the-huffington-post-18468761129.csv%2Fthe-huffington-post-18468761129.csv?auth=eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJwcm9kLXVzZXItY2xpZW50OnFpYW5oZW5nIiwiaXNzIjoiYWdlbnQ6cWlhbmhlbmc6OmUwN2U0NmE4LWRlMjQtNGZlZi1hMDQ3LTBlYjUzZGE3MTNjZSIsImlhdCI6MTU4NTAzNTgxOSwicm9sZSI6WyJ1c2VyIiwidXNlcl9hcGlfYWRtaW4iLCJ1c2VyX2FwaV9yZWFkIiwidXNlcl9hcGlfd3JpdGUiXSwiZ2VuZXJhbC1wdXJwb3NlIjpmYWxzZSwidXJsIjoiYTM3MmJkZTk3OGRlNzU4NmFlYWFkZGEwZjNjM2UxY2ZlNGM3MGVlNCJ9.KWt8-lcb5W73WJXUyQVclYbkCfcvdm2rmvqaj6iUKqsHlc6l4390JXuwtngOkcXiWho708N7r-5KSFAjf-aSYQ [following]
--2020-03-24 15:31:52--  https://d

In [58]:
!gunzip -c abc_news_86680728811 > abc_news_86680728811.csv
!rm abc_news_86680728811
!gunzip -c bbc_228735667216 > bbc_228735667216.csv
!rm bbc_228735667216
!gunzip -c cbs_news_131459315949 > cbs_news_131459315949.csv
!rm cbs_news_131459315949
!gunzip -c cnn_5550296508 > cnn_5550296508.csv
!rm cnn_5550296508
!gunzip -c fox_and_friends_111938618893743 > fox_and_friends_111938618893743.csv
!rm fox_and_friends_111938618893743
!gunzip -c fox_news_15704546335 > fox_news_15704546335.csv
!rm fox_news_15704546335
!gunzip -c nbc_news_155869377766434 > nbc_news_155869377766434.csv
!rm nbc_news_155869377766434
!gunzip -c npr_10643211755 > npr_10643211755.csv
!rm npr_10643211755
!gunzip -c the_huffington_post_18468761129 > the_huffington_post_18468761129.csv
!rm the_huffington_post_18468761129
!gunzip -c the_los_angeles_times_5863113009 > the_los_angeles_times_5863113009.csv
!rm the_los_angeles_times_5863113009

!unzip the_new_york_times_5281959998
!rm the_new_york_times_5281959998
!mv the-new-york-times-5281959998.csv the_new_york_times_5281959998.csv
!unzip the_wall_street_journal_8304333127
!rm the_wall_street_journal_8304333127
!mv the-wall-street-journal-8304333127.csv the_wall_street_journal_8304333127.csv
!unzip the_washington_post_6250307292
!rm the_washington_post_6250307292
!mv the-washington-post-6250307292.csv the_washington_post_6250307292.csv
!unzip time_10606591490
!rm time_10606591490
!mv time-10606591490.csv time_10606591490.csv
!unzip usa_today_13652355666
!rm usa_today_13652355666
!mv usa-today-13652355666.csv usa_today_13652355666.csv

Archive:  the_new_york_times_5281959998
  inflating: the-new-york-times-5281959998.csv  
   creating: __MACOSX/
  inflating: __MACOSX/._the-new-york-times-5281959998.csv  
Archive:  the_wall_street_journal_8304333127
  inflating: the-wall-street-journal-8304333127.csv  
  inflating: __MACOSX/._the-wall-street-journal-8304333127.csv  
Archive:  the_washington_post_6250307292
  inflating: the-washington-post-6250307292.csv  
  inflating: __MACOSX/._the-washington-post-6250307292.csv  
Archive:  time_10606591490
  inflating: time-10606591490.csv    
  inflating: __MACOSX/._time-10606591490.csv  
Archive:  usa_today_13652355666
  inflating: usa-today-13652355666.csv  
  inflating: __MACOSX/._usa-today-13652355666.csv  


In [59]:
!iconv -f UTF-16 -t UTF-8 abc_news_86680728811.csv > /home/qianhengma/Documents/BDA_TP_1/data/abc_news_86680728811_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 bbc_228735667216.csv > /home/qianhengma/Documents/BDA_TP_1/data/bbc_228735667216_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 cbs_news_131459315949.csv > /home/qianhengma/Documents/BDA_TP_1/data/cbs_news_131459315949_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 cnn_5550296508.csv > /home/qianhengma/Documents/BDA_TP_1/data/cnn_5550296508_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 fox_and_friends_111938618893743.csv > /home/qianhengma/Documents/BDA_TP_1/data/fox_and_friends_111938618893743_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 fox_news_15704546335.csv > /home/qianhengma/Documents/BDA_TP_1/data/fox_news_15704546335_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 nbc_news_155869377766434.csv > /home/qianhengma/Documents/BDA_TP_1/data/nbc_news_155869377766434_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 npr_10643211755.csv > /home/qianhengma/Documents/BDA_TP_1/data/npr_10643211755_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 the_huffington_post_18468761129.csv > /home/qianhengma/Documents/BDA_TP_1/data/the_huffington_post_18468761129_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 the_los_angeles_times_5863113009.csv > /home/qianhengma/Documents/BDA_TP_1/data/the_los_angeles_times_5863113009_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 the_new_york_times_5281959998.csv > /home/qianhengma/Documents/BDA_TP_1/data/the_new_york_times_5281959998_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 the_wall_street_journal_8304333127.csv > /home/qianhengma/Documents/BDA_TP_1/data/the_wall_street_journal_8304333127_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 the_washington_post_6250307292.csv > /home/qianhengma/Documents/BDA_TP_1/data/the_washington_post_6250307292_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 time_10606591490.csv > /home/qianhengma/Documents/BDA_TP_1/data/time_10606591490_UTF_8.csv
!iconv -f UTF-16 -t UTF-8 usa_today_13652355666.csv > /home/qianhengma/Documents/BDA_TP_1/data/usa_today_13652355666_UTF_8.csv

In [63]:
ll data/

total 370824
-rw-rw-r-- 1 qianhengma 26995316 Mar 24 15:48 abc_news_86680728811_UTF_8.csv
-rw-rw-r-- 1 qianhengma 11542367 Mar 24 15:48 bbc_228735667216_UTF_8.csv
-rw-rw-r-- 1 qianhengma 25530449 Mar 24 15:48 cbs_news_131459315949_UTF_8.csv
-rw-rw-r-- 1 qianhengma 19591227 Mar 24 15:48 cnn_5550296508_UTF_8.csv
-rw-rw-r-- 1 qianhengma  3390449 Mar 24 15:48 fox_and_friends_111938618893743_UTF_8.csv
-rw-rw-r-- 1 qianhengma 21795606 Mar 24 15:48 fox_news_15704546335_UTF_8.csv
-rw-rw-r-- 1 qianhengma 30891664 Mar 24 15:48 nbc_news_155869377766434_UTF_8.csv
-rw-rw-r-- 1 qianhengma 31802002 Mar 24 15:48 npr_10643211755_UTF_8.csv
-rw-rw-r-- 1 qianhengma 59461978 Mar 24 15:48 the_huffington_post_18468761129_UTF_8.csv
-rw-rw-r-- 1 qianhengma 23131997 Mar 24 15:48 the_los_angeles_times_5863113009_UTF_8.csv
-rw-rw-r-- 1 qianhengma 35190504 Mar 24 15:48 the_new_york_times_5281959998_UTF_8.csv
-rw-rw-r-- 1 qianhengma 27325397 Mar 24 15:48 the_wall_street_journal_8304333127_UTF_8.csv
-rw-rw-r-- 1 qia

In [64]:
!hdfs dfs -put /home/qianhengma/Documents/BDA_TP_1/data/*.csv /user/input/

20/03/24 15:50:06 WARN hdfs.DataStreamer: Caught exception
java.lang.InterruptedException
	at java.lang.Object.wait(Native Method)
	at java.lang.Thread.join(Thread.java:1252)
	at java.lang.Thread.join(Thread.java:1326)
	at org.apache.hadoop.hdfs.DataStreamer.closeResponder(DataStreamer.java:973)
	at org.apache.hadoop.hdfs.DataStreamer.endBlock(DataStreamer.java:624)
	at org.apache.hadoop.hdfs.DataStreamer.run(DataStreamer.java:801)


In [65]:
!hdfs dfs -ls /user/input/

Found 15 items
-rw-r--r--   3 qianhengma supergroup   26995316 2020-03-24 15:50 /user/input/abc_news_86680728811_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   11542367 2020-03-24 15:50 /user/input/bbc_228735667216_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   25530449 2020-03-24 15:50 /user/input/cbs_news_131459315949_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   19591227 2020-03-24 15:50 /user/input/cnn_5550296508_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup    3390449 2020-03-24 15:50 /user/input/fox_and_friends_111938618893743_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   21795606 2020-03-24 15:50 /user/input/fox_news_15704546335_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   30891664 2020-03-24 15:50 /user/input/nbc_news_155869377766434_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   31802002 2020-03-24 15:50 /user/input/npr_10643211755_UTF_8.csv
-rw-r--r--   3 qianhengma supergroup   59461978 2020-03-24 15:50 /user/input/the_huffington_post_18468761129_UTF_8.csv
-

In [67]:
!pip3 install pyhdfs

    100% |████████████████████████████████| 61kB 1.2MB/s ta 0:00:011
    100% |████████████████████████████████| 92kB 6.2MB/s eta 0:00:01
    100% |████████████████████████████████| 61kB 6.6MB/s eta 0:00:01
    100% |████████████████████████████████| 133kB 5.4MB/s eta 0:00:01
    100% |████████████████████████████████| 163kB 1.9MB/s ta 0:00:011
    100% |████████████████████████████████| 143kB 5.1MB/s eta 0:00:01
  Running setup.py bdist_wheel for pyhdfs ... done
  Stored in directory: /home/qianhengma/.cache/pip/wheels/05/e8/fa/efdb743f2c14cda7f85d258ef270a591d1b2f00473aaf494dd
  Running setup.py bdist_wheel for simplejson ... done
  Stored in directory: /home/qianhengma/.cache/pip/wheels/86/c0/83/dcd0339abb2640544bb8e0938aab2d069cef55e5647ce6e097
Successfully built pyhdfs simplejson


In [6]:
import pyhdfs

In [7]:
hdfs = pyhdfs.HdfsClient(hosts='cluster001:50070', user_name='qianhengma')

In [8]:
files_name = hdfs.listdir('/user/input/')

In [9]:
print(files_name)
print(len(files_name))

['abc_news_86680728811_UTF_8.csv', 'bbc_228735667216_UTF_8.csv', 'cbs_news_131459315949_UTF_8.csv', 'cnn_5550296508_UTF_8.csv', 'fox_and_friends_111938618893743_UTF_8.csv', 'fox_news_15704546335_UTF_8.csv', 'nbc_news_155869377766434_UTF_8.csv', 'npr_10643211755_UTF_8.csv', 'the_huffington_post_18468761129_UTF_8.csv', 'the_los_angeles_times_5863113009_UTF_8.csv', 'the_new_york_times_5281959998_UTF_8.csv', 'the_wall_street_journal_8304333127_UTF_8.csv', 'the_washington_post_6250307292_UTF_8.csv', 'time_10606591490_UTF_8.csv', 'usa_today_13652355666_UTF_8.csv']
15


In [10]:
count = 0
final_data = ''
for file_name in files_name:
    inputfile_dir = 'hdfs://cluster001:9000/user/input/' + file_name
    tmp_rdd = handle_csv(inputfile_dir)
    if count==0:
        final_data = tmp_rdd
    else:
        final_data = final_data.union(tmp_rdd)
    count += 1

In [11]:
final_data.count()

534391

In [14]:
stop_words=set(stopwords.words('english'))
stop_words.add('http')
stop_words.add('https')
stop_words.add('null')
list_punct=list(string.punctuation) + ['’','“','”','‘','—']
key_words = ['Thai', 'Thailand', 'thai', 'thailand']

In [32]:
def contains_any(arr):
	for element in arr:
		if isinstance(element, str):
			for key_word in key_words:
				if key_word in element:
					return True
		else :
			for key_word in key_words:
				if key_word in str(element):
					return True
	return False

In [33]:
classified_text_rdd = final_data \
.map(lambda line :line) \
.filter(contains_any) \
.flatMap(lambda row: row) \
.flatMap(word_tokenize) \
.map(lambda word: word.lower()) \
.filter(lambda word : word not in stop_words and word != '') \
.filter(lambda word : word not in list_punct) \
.map(pos_tag) \
.filter(lambda word : word[0][1][0] == 'N') \
.map(lambda word: word[0][0]) \
.map(lambda word: (word, 1)) \
.reduceByKey(lambda a, b: a+b) \
.sortBy(lambda a: a[1], ascending = False)

In [34]:
# print(classified_text_rdd.collect())
classified_text_rdd \
.saveAsTextFile('hdfs://cluster001:9000/user/output/BDA_TP_1_b')

In [35]:
!hdfs dfs -ls /user/output/BDA_TP_1_b

Found 30 items
-rw-r--r--   3 qianhengma supergroup          0 2020-03-25 19:47 /user/output/BDA_TP_1_b/_SUCCESS
-rw-r--r--   3 qianhengma supergroup       1359 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00000
-rw-r--r--   3 qianhengma supergroup       2153 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00001
-rw-r--r--   3 qianhengma supergroup       1322 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00002
-rw-r--r--   3 qianhengma supergroup       2813 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00003
-rw-r--r--   3 qianhengma supergroup          0 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00004
-rw-r--r--   3 qianhengma supergroup       3257 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00005
-rw-r--r--   3 qianhengma supergroup          0 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00006
-rw-r--r--   3 qianhengma supergroup          0 2020-03-25 19:47 /user/output/BDA_TP_1_b/part-00007
-rw-r--r--   3 qianhengma supergroup       5210 2020-03-25 19:47 /user/output/BDA_TP_1_

In [36]:
!hdfs dfs -cat /user/output/BDA_TP_1_b/part-00000

('thailand', 615)
('thai', 356)
('bangkok', 128)
('king', 101)
('police', 101)
('people', 85)
('photos', 76)
('country', 73)
('world', 73)
('temple', 68)
('timeline', 55)
('tourists', 54)
('found', 51)
('man', 51)
('tiger', 49)
('bhumibol', 47)
('elephant', 45)
('coup', 41)
('years', 41)
('shrine', 39)
('tourist', 38)
('thais', 38)
('buddhist', 37)
('asia', 36)
('bombing', 35)
('death', 35)
('minister', 33)
('beach', 32)
('officials', 32)
('government', 31)
('myanmar', 30)
('prime', 30)
('city', 29)
('army', 29)
('authorities', 29)
('bomb', 28)
('baby', 28)
('suspect', 28)
('time', 28)
('day', 28)
('family', 27)
('show', 27)
('adulyadej', 27)
('year', 27)
('constitution', 27)
('island', 27)
('malaysia', 27)
('monks', 27)
('court', 26)
('food', 26)
('koh', 25)
('dog', 25)
('junta', 25)
('photo', 24)
('law', 24)
('china', 23)
('news', 23)
('yingluck', 23)
('way', 23)
('watch', 22)
('tourism', 22)
('women', 22)
('cubs', 21)
('facebook', 21)
('tigers', 21)
('shinawatra', 21)
('caught', 20)

In [37]:
output_files_name = hdfs.listdir('/user/output/BDA_TP_1_b')

In [38]:
print(output_files_name)

['_SUCCESS', 'part-00000', 'part-00001', 'part-00002', 'part-00003', 'part-00004', 'part-00005', 'part-00006', 'part-00007', 'part-00008', 'part-00009', 'part-00010', 'part-00011', 'part-00012', 'part-00013', 'part-00014', 'part-00015', 'part-00016', 'part-00017', 'part-00018', 'part-00019', 'part-00020', 'part-00021', 'part-00022', 'part-00023', 'part-00024', 'part-00025', 'part-00026', 'part-00027', 'part-00028']


In [39]:
!hdfs dfs -mkdir -p /user/output/BDA_TP_1_b_merged

In [2]:
# merge part files
!hdfs dfs -cat /user/output/BDA_TP_1_b/* | hdfs dfs -put - /user/output/BDA_TP_1_b_merged/BDA_TP_1_b_output.txt

In [3]:
!hdfs dfs -cat /user/output/BDA_TP_1_b_merged/BDA_TP_1_b_output.txt

('thailand', 615)
('thai', 356)
('bangkok', 128)
('king', 101)
('police', 101)
('people', 85)
('photos', 76)
('country', 73)
('world', 73)
('temple', 68)
('timeline', 55)
('tourists', 54)
('found', 51)
('man', 51)
('tiger', 49)
('bhumibol', 47)
('elephant', 45)
('coup', 41)
('years', 41)
('shrine', 39)
('tourist', 38)
('thais', 38)
('buddhist', 37)
('asia', 36)
('bombing', 35)
('death', 35)
('minister', 33)
('beach', 32)
('officials', 32)
('government', 31)
('myanmar', 30)
('prime', 30)
('city', 29)
('army', 29)
('authorities', 29)
('bomb', 28)
('baby', 28)
('suspect', 28)
('time', 28)
('day', 28)
('family', 27)
('show', 27)
('adulyadej', 27)
('year', 27)
('constitution', 27)
('island', 27)
('malaysia', 27)
('monks', 27)
('court', 26)
('food', 26)
('koh', 25)
('dog', 25)
('junta', 25)
('photo', 24)
('law', 24)
('china', 23)
('news', 23)
('yingluck', 23)
('way', 23)
('watch', 22)
('tourism', 22)
('women', 22)
('cubs', 21)
('facebook', 21)
('tigers', 21)
('shinawatra', 21)
('caught', 20)

# Summary
    The word Thailand appears 615 times, followed by Thai at 356, Bangkok at 128, King at 101. It may shows the capital Bangkok is very active and famous in Thailand. The next words are temple at 68, tourists at 54, Bhumibol at 47, elephant at 45, coup at 41, shrine at 39. 
    It can show that there are many temples and tourists in Thailand. Many people believe in Buddhist religion.
    On the lower numbers, Royal was used 17 times, Prince at 13, protests at 12, beaches at 10, Leicester at 9, and monk at 8.
    On the least used words, Phuket was used 6 times, politics at 5, Songkran at 3, Thaksin at 3, massage at 2, Erawan shrine at 2, and religion at 1.